src/main/java/controller/ListChats.java

```diff
@@ -160,12 +163,16 @@ public void listChatsWithoutUserIdPathVariable(HttpServletRequest request) {
         }
 
         //Stores list of all details of each Chat of the User.
-        List<Map<String, Object>> chatInfoOfChatsOfUser = new ArrayList<Map<String, Object>>();
+        List<ImmutableMap<String, Object>> chatInfoOfChatsOfUser = new ArrayList<ImmutableMap<String, Object>>();
 
         for (Chat chat : chatsOfUser) {
             chatInfoOfChatsOfUser.add(getChatInfoOfChatInMap(chat, chatIdSecondUsernameMap.get(chat.getChatId())));
         }
 
-        return SuccessResponseGenerator.getSuccessResponseForListChats(chatInfoOfChatsOfUser);
+        ImmutableList<ImmutableMap<String, Object>> chatInfoOfChatsOfUserImmutable = ImmutableList.<ImmutableMap<String, Object>> builder()
+                                                                                                .addAll(chatInfoOfChatsOfUser)
+                                                                                                .build();
```

Comment:
Instead of adding to a mutable list and then copying over to an immutable one, you could directly add to the `ImmutableList.Builder`.

Reply:
Done